static/js/popovers.js

```diff
@@ -380,7 +433,8 @@ export function show_user_profile(user) {
     const profile_data = page_params.custom_profile_fields
         .map((f) => get_custom_profile_field_data(user, f, field_types, dateFormat))
         .filter((f) => f.name !== undefined);
-
+    const user_streams = stream_data.subscribed_subs();
+    const groups_of_user = user_groups.get_user_groups_of_user(user.user_id);
```

Comment:
It seems `subscribed_subs` returns the streams the current user has subscribed to? (As opposed to that of `user.user_id`?)

Reply:
Hmm this is the reason I was seeing same stream list for almost all profile modals. Thanks for reporting, I'll fix it, actually it was a part of commits from abhirup's PR which I did not changed(assuming that it was reviewed) .

Reply:
Good find.

Generally, you shouldn't assume that code in a previous PR is correct or reviewed; it's always best to understand all of the code yourself when picking up an existing project.